core/src/main/java/com/datastax/oss/driver/api/core/metadata/schema/TableMetadata.java

```diff
@@ -32,6 +32,10 @@
   @NonNull
   @Override
   default String describe(boolean pretty) {
+    // this is a virtual table, do not return a describe string
+    if (getPrimaryKey().size() == 0) {
+      return "";
+    }
```

Comment:
We should have an `isVirtual()` method on `TableMetadata` as well.

And, even though the PK size is probably a fairly reliable indicator, it would be cleaner to store the information explicitly, as a `virtual` field that gets passed to `DefaultTableMetadata`'s constructor. It will be easy to set since you already have a separate method to build virtual tables in `TableParser`.